apps/rush-lib/src/api/RushProjectConfiguration.ts

```diff
@@ -21,6 +21,15 @@ interface IRushProjectJson {
    */
   projectOutputFolderNames?: string[];
 
+  /**
+   * A list of file globs under the project root that will be ignored when
+   * determining whether this project has changed.
+   */
+  ignoreChangedFiles?: string[];
```

Comment:
The phrase "changed files" sounds like it is somehow related to the `rush change` feature, which this isn't.

If they are globs, we should call them "globs".  (Heft has some settings like ["includeGlobs" and "excludeGlobs"](https://rushstack.io/pages/heft_tasks/copy-files/) for example.)

You suggested `ignoreFilePatternsForBuildState`, but that sounds limited to `rush build`, which this isn't.  The incremental build analysis can be enabled for various Rush **commands**, which includes `rush build` but also [custom commands](https://rushjs.io/pages/maintainer/custom_commands/).  A user might define an incremental command like `rush generate-docs` and then enable [this setting](https://rushjs.io/pages/configs/command-line_json/):

```
    //   /**
    //    * If true then this command will be incremental like the built-in "build" command
    //    */
    //   "incremental": false,
```
(We could provide separate ignore globs for each command, but that seems like an advanced feature that will get entangled with other design problems like [phased commands](https://github.com/microsoft/rushstack/issues/2300).)  My point is that "incremental" or "incremental build" is perhaps more clear than "build" by itself.

So based on all this, maybe the setting could be something like:
- `incrementalIgnoredGlobs` 
- `incrementalBuildIgnoredGlobs`
- `incrementalCommandIgnoredGlobs`

@iclanton what do you think?

Reply:
I think `incrementalBuildIgnoredGlobs` is just about right.  The phrase "incremental build" is the most common phrasing by far, in my opinion, so it makes sense to say "here's ignored globs for incremental build" (the fact that it applies to other commands potentially is OK, since those commands "act like incremental build" -- that's the way you'd describe them to someone).